### DISCUSSION TITLE
Attempting to run a search from Ruby

### DISCUSSION THREAD

**Discussion by jeffkees1** [milvus-io/milvus#16407] [Q&A and General discussion] [answered]
We are making requests into Milvus from a Rails application.  We have been able to issue a few simple commands like show_collections and show_partitions  The results looked good.

Now we're trying to run a search.  Reverse engineering the python client has given us most of what I need to run a search.  However, we have run into a piece of code that does not seem to have a ruby equivalent.  The PlaceholderGroup field for the SearchRequest must be serialized.  The python client does it like this:
`plg_str = milvus_types.PlaceholderGroup.SerializeToString(plg)`

The ruby version of PlaceholderGroup does not seem to have a `serialize_to_string` method.  Is there another way to serialize a PlaceholderGroup from Ruby?

**Comment by yhmo** [comment]
The SerializeToString() is a method of protobuf lib. The method name can be different in different language.
In C++ sdk, the method name is "SerializeAsString":
```
        proto::milvus::PlaceholderGroup placeholder_group;
        search_request.set_placeholder_group(std::move(placeholder_group.SerializeAsString()));
```

In Java sdk, the method name is "toByteString":
```
        PlaceholderGroup placeholderGroup = PlaceholderGroup.newBuilder()
                .addPlaceholders(plv)
                .build();

        ByteString byteStr = placeholderGroup.toByteString();
        search_request_builder.setPlaceholderGroup(byteStr);
```

In Ruby, there must be an equivalent method.

**Reply by jeffkees1** [reply]
Thank you for your reply.  I think the method may be: PlaceholderGroup.encode(proto)

I'm trying to verify this is correct.

**Reply by xiaofan-luan** [reply]
> Thank you for your reply. I think the method may be: PlaceholderGroup.encode(proto)
> 
> I'm trying to verify this is correct.

Not an expert on ruby, but glad to see sb is working on it. Thanks @jeffkees1

**Comment by jeffkees1** [comment, marked as answer]
Verified:  PlaceholderGroup.encode(plg)  is the correct ruby method to serialize a PlaceholderGroup for a SearchRequest.
